recipes/cpp-sort/all/conanfile.py

```diff
@@ -15,6 +15,9 @@ class CppSortConan(ConanFile):
 
     _source_subfolder = "source_subfolder"
 
+    def configure(self):
+        tools.check_min_cppstd(self, "14")
```

Comment:
Sadly this on it's own is not enough the `cppstd` setting may or omay be present ( it's something thats come up in the past ) you will need to do something more complex like https://github.com/conan-io/conan-center-index/blob/f8f8950c8b87bb5061f082fef6f444a6579bc545/recipes/ms-gsl/all/conanfile.py#L47-L54

Reply:
> Now I'm not sure how to get that one to work: GCC 4.9 can't compile the library because it doesn't support variable templates, but I'm unsure how to express that information in the Conan model. Might someone help?

In general the answer is to raise an `ConanInvalidConfiguration`. For example, a project's options conflicting maybe be a reason. This case, which is common, a minimum compiler is required so we throw an exception when we know, otherwise flag a warning and with the use luck =)

Reply:
I mean, I knew about raising `ConanInvalidConfiguration`, but I thought that there could be a more user-friendly way than to manually specify all minimum compiler versions by hand for this specific problem :/

Anyway, thanks for the information, I'll probably copy-paste one of the snippets you linked and adjust as needed.

Reply:
It's on my todo list to open an issue against the conan repo I completely agree.